src/Core/System/Snippet/Command/Util/CountryAgnosticFileValidator.php

```diff
@@ -0,0 +1,175 @@
+        }
+
+        return $finder;
+    }
+
+    public function checkTranslationFiles(ValidatedTranslationFileOptions $options): ValidatedTranslationFileStruct
+    {
+        $finder = $this->getFinder($options);
+        if ($finder->count() < 0) {
+            return new ValidatedTranslationFileStruct();
+        }
+
+        $languageFiles = new TranslationFileCollection([]);
+        $countrySpecificFileCollection = new TranslationFileCollection([]);
+        foreach ($finder as $file) {
+            $filename = $file->getFilename();
+
+            $isCoreTranslationFile = preg_match(
+                SnippetPatterns::CORE_SNIPPET_FILE_PATTERN,
+                $filename,
+                $coreFileData,
+                \PREG_UNMATCHED_AS_NULL
+            );
+
+            $isAdminTranslationFile = preg_match(
+                SnippetPatterns::ADMIN_SNIPPET_FILE_PATTERN,
+                $filename,
+                $adminFileData,
+                \PREG_UNMATCHED_AS_NULL
+            );
+
+            if (!$isAdminTranslationFile && !$isCoreTranslationFile) {
+                continue;
+            }
+
+            $currentFileData = $isAdminTranslationFile ? $adminFileData : $coreFileData;
+
+            $currentDomain = $currentFileData['domain'] ?? 'administration';
+            $formatedLanguageStruct = new TranslationFile(
+                $filename,
+                $file->getPath(),
+                $currentDomain,
+                str_replace('_', '-', $currentFileData['locale']),
+                $currentFileData['language'] ?? null,
+                $currentFileData['script'] ?: null,
+                $currentFileData['region'] ?: null,
+                !$isAdminTranslationFile && !empty($currentFileData['isBase']),
+            );
+
+            $fileIdentifier = $formatedLanguageStruct->getFullPath();
+
+            if ($formatedLanguageStruct->getRegion()) {
+                $countrySpecificFileCollection->set($fileIdentifier, $formatedLanguageStruct);
+            }
+
+            $languageFiles->set($fileIdentifier, $formatedLanguageStruct);
+        }
+
+        return $this->processAgnosticFiles(new ValidatedTranslationFileStruct(
+            $languageFiles,
+            $countrySpecificFileCollection,
+        ));
+    }
+
+    public function processAgnosticFiles(ValidatedTranslationFileStruct $validatedFileStruct): ValidatedTranslationFileStruct
+    {
+        $count = $validatedFileStruct->getSpecificCollection()->count();
+        if ($count < 1) {
+            return $validatedFileStruct;
+        }
+
+        foreach ($validatedFileStruct->getSpecificCollection()->getElements() as $countrySpecificFile) {
+            $translationCollection = $validatedFileStruct->getCompleteCollection();
+
+            // If no agnostic file exists, $countrySpecificFile is content for `fixFilenames` to be fixed
+            if ($translationCollection->get($countrySpecificFile->getAgnosticPath()) === null) {
```

Comment:
how do you ensure, that the `TranslationFileCollection` always uses the agnostic path as key? 

I needed something similar. you might want to have a look https://github.com/shopware/shopware/blob/trunk/src/Core/Content/Cookie/Struct/CookieGroupCollection.php

Reply:
Good catch. I changed it to "add" and overrode it with a set using its fullpath. This will exactly match the usecase.

The agnosticPath here is only used because it's a complete collection with all snippet files 🙂